### PR TITLE
CA-260262: Restrict allowed VDI operations during rolling pool upgrade to those supported by older releases

### PR DESCRIPTION
**Pull request by gaborigloi**

### CODE REVIEW

**Comment by gaborigloi** [comment]
Is CA-260262 the ticket for this issue? I should probably include it in the title & commit messages.

**Comment by gaborigloi** [comment]
I assume that's the right issue, changed the commit message titles.

File: ocaml/xapi/xapi_globs.ml [hunk 1]
```diff
@@ -454,6 +454,21 @@ let rpu_allowed_vm_operations = [
   `update_allowed_operations;
 ]
 
+let rpu_allowed_vdi_operations = [
+  `clone;
+  `copy;
+  `resize;
+  `resize_online;
+  `snapshot;
+  `mirror;
```
Comment (robhoes) [inline ocaml/xapi/xapi_globs.ml]: Mirror is here, but removed again in the next patch.

File: ocaml/xapi/test_vdi_allowed_operations.ml [hunk 2]
```diff
@@ -274,23 +274,42 @@ let test_cbt =
...
+  in
 
+  let test_update_allowed_operations () =
+    let __context = Mock.make_context_with_new_db "Mock context" in
+    let master = Test_common.make_host __context () in
+    let pool = Test_common.make_pool ~__context ~master () in
+    Db.Pool.add_to_other_config ~__context ~self:pool ~key:Xapi_globs.rolling_upgrade_in_progress ~value:"x";
+    let self, _ = setup_test ~__context ~vdi_fun:(fun vdi -> Db.VDI.set_type ~__context ~self:vdi ~value:`user) () in
+    Xapi_vdi.update_allowed_operations ~__context ~self;
+    OUnit.assert_bool "update_allowed_operations should exclude `enable_cbt during RPU" (not @@ List.mem `set_on_boot (Db.VDI.get_allowed_operations ~__context ~self));
```
Comment (robhoes) [inline ocaml/xapi/test_vdi_allowed_operations.ml]: The string says "enable_cbt" but the code says "set_on_boot"...? Below also.

**Comment by gaborigloi** [comment]
Thanks for the review, I've fixed the typos and the commits.

**Pull request merged**